packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -1345,6 +1348,41 @@ export function listenToEventResponderEventTypes(
   }
 }
 
+export function listenToListenerSystemEvent(
+  type: string,
+  passive: boolean,
+): void {
+  if (enableListenerAPI) {
+    // Get the listening Map for this element. We use this to track
+    // what events we're listening to.
+    const listenerMap = getListenerMapForElement(window);
+    const passiveKey = type + '_passive';
+    const activeKey = type + '_active';
+    const eventKey = passive ? passiveKey : activeKey;
+
+    if (!listenerMap.has(eventKey)) {
+      if (passive) {
+        if (listenerMap.has(activeKey)) {
+          // If we have an active event listener, do not register
+          // a passive event listener. We use the same active event
+          // listener.
+          return;
+        } else {
+          // If we have a passive event listener, remove the
+          // existing passive event listener before we add the
+          // active event listener.
+          const passiveListener = listenerMap.get(passiveKey);
+          if (passiveListener != null) {
```

Comment:
Shouldn't this be `passiveListener !== null`?

Reply:
Nope. If the key doesn’t exist then the passiveListener is always undefined. This could probably change from being a nullish check to being a void check.